### ISSUE TITLE
Performance improvements for large time series 

### ISSUE THREAD

**Issue by dflatow** [heartexlabs/label-studio#2118]
### Is your feature request related to a problem?

Performance degradation with larger time-series. 

### Describe the solution you'd like

We work with very large time series which can be highly demanding when fully loaded in memory. We would like to be able to be able to: 

- Open the large file and load a standard field of view (frame)
- While also lazy loading a buffer of previous and subsequent frames
- Without necessarily loading the entire file

As the user moves forward or backward in the time series we lazy load the corresponding previous and subsequent frames

### A clear and concise description of any alternative solutions or features you've considered.

Break down the large files in much smaller files and load each one separately which makes data handling and labels handling very cumbersome.

### Add any other context or screenshots about the feature request here.

I think this is sufficiently different from https://github.com/heartexlabs/label-studio/issues/1326 to warrant a new issue.

**Comment by makseq** [comment]
@dflatow Thank you very much for your feature request ;-)

**Comment by pidefrem** [comment]
I agree, lazy loading of some kind would be a huge improvement for entreprise adoption on real use cases. We are currently facing a lot of performance problems on our time series use case especially when users don't have the same connection speed as developers. For example, users get timeouts and can't load the project. Is there something on your roadmap to tackle this problem?
